homeassistant/components/xiaomi_miio/gateway.py

```diff
@@ -33,16 +42,45 @@ def gateway_info(self):
     async def async_connect_gateway(self, host, token):
         """Connect to the Xiaomi Gateway."""
         _LOGGER.debug("Initializing with host %s (token %s...)", host, token[:5])
+
+        use_cloud = self._config_entry.options.get(CONF_CLOUD_SUBDEVICES)
```

Comment:
We need to have defaults here unless we set defaults in the entry options during config entry setup. The options flow won't be run by default on entry setup.

Reply:
The cloud settings/credentials are all optional, if they are not specified this gives None and that means the cloud is not used and only local commands are used. The default is therefore None which is now handeld, but I will make the default False, which makes it a bit more clear.